pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -552,7 +552,7 @@ func ResolvePipelineTask(
 		numCombinations = pipelineTask.Matrix.CountCombinations()
```

Comment:
I'm curious what `CountCombinations` returns here for full array references? It seems like it would not return the correct answer, as the number of combinations is not yet known. If this is the case, what does rpt.TaskRunNames get set to, and likewise for RunObjectNames?